From 87c158cfc2c0e42e9acf5670dc976750fc0974b0 Mon Sep 17 00:00:00 2001
From: Guo Mengqi <guomengqi3@huawei.com>
Date: Thu, 20 Oct 2022 16:23:56 +0800
Subject: [PATCH 025/595] [Huawei] Fix CodeCheck2.0 static check warning

Offering: HULK
hulk inclusion
category: bugfix
bugzilla: 187808

-----------------------------------------------

fix following CodeCheck warnings:

(1) Avoid nesting a code block more than four times within a function.
    Function nesting depth is 5. Function name is : remove_inode_hugepages.
(2) The operand type of unary operator '!' is improperly 'int'

Signed-off-by: Guo Mengqi <guomengqi3@huawei.com>
---
 fs/hugetlbfs/inode.c | 12 +++++-------
 1 file changed, 5 insertions(+), 7 deletions(-)

diff --git a/fs/hugetlbfs/inode.c b/fs/hugetlbfs/inode.c
index 379aa008514d..ba19eab8a74e 100644
--- a/fs/hugetlbfs/inode.c
+++ b/fs/hugetlbfs/inode.c
@@ -601,13 +601,11 @@ static void remove_inode_hugepages(struct inode *inode, loff_t lstart,
 			 * if the page is from buddy system, do not add to freed.
 			 * because freed is used for hugetlbfs reservation accounting.
 			 */
-			if (!HPageTemporary(page)) {
+			if (HPageTemporary(page) == 0) {
 				freed++;
-				if (!truncate_op) {
-					if (unlikely(hugetlb_unreserve_pages(inode,
-									index, index + 1, 1)))
-						hugetlb_fix_reserve_counts(inode);
-				}
+				if (!truncate_op && unlikely(hugetlb_unreserve_pages(inode,
+								index, index + 1, 1)))
+					hugetlb_fix_reserve_counts(inode);
 			}
 
 			unlock_page(page);
@@ -1060,7 +1058,7 @@ static int hugetlbfs_error_remove_page(struct address_space *mapping,
 	pgoff_t index = page->index;
 
 	remove_huge_page(page);
-	if (!HPageTemporary(page) &&
+	if (HPageTemporary(page) == 0 &&
 			unlikely(hugetlb_unreserve_pages(inode, index, index + 1, 1)))
 		hugetlb_fix_reserve_counts(inode);
 
-- 
2.17.1

